search/xml.py

```diff
@@ -0,0 +1,50 @@
+from io import BytesIO, StringIO
+from typing import Union
+
+from lxml import isoschematron
+from lxml.etree import DTD, ElementTree, RelaxNG, XMLSchema, parse
+
+_VALIDATORS = {
+    'dtd': DTD,
+    'rng': RelaxNG,
+    'relaxng': RelaxNG,
+    'xsd': XMLSchema,
+    'xmlschema': XMLSchema,
+    'sch': isoschematron.Schematron,
+    'schematron': isoschematron.Schematron
+}
+
+
+class UnsupportedSchemaException(Exception):
+    pass
+
+
+def parse_xml_from_bytes(xml_bytes) -> ElementTree:
+    return parse(BytesIO(xml_bytes))
+
+
+def parse_xml_from_string(xml_string) -> ElementTree:
+    return parse(StringIO(xml_string))
+
+
+def parse_xml_file(file_path: str) -> ElementTree:
+    with open(file_path, 'rb') as file:
+        data = file.read()
+    return parse_xml_from_bytes(data)
+
+
+def get_validator(schema_name: str) -> Union[DTD, RelaxNG, XMLSchema,
+                                             isoschematron.Schematron]:
```

Comment:
`_Validator` as a simpler return type?

Reply:
I have since removed this file as these functions are no longer needed.